sycl/source/detail/graph_impl.hpp

```diff
@@ -313,6 +340,16 @@ class graph_impl {
     MInorderQueueMap[QueueWeakPtr] = Node;
   }
 
+  /// Prints contents of graph to text file in DOT format
+  /// \param GraphName is a string appended to the output file name
+  void printGraphAsDot(const char *GraphName);
```

Comment:
should be `const` method

Reply:
I don't think so, as this function modifies the MVisited member of visited nodes.

Reply:
I think we could get rid of the `MVisited` member and rather than doing this in `printDotRecursive`
```c++
    if (MVisited)
      return;
    MVisited = true;
    Visited.push_back(this);
```
Just check for the early return if `this` is already in `Visited`.
```cpp
if (std::find(Visited.begin(), Visited.end(), this) == Visited.end())
  return;

 Visited.push_back(this);
  
```

Doing this lookup every time might be slower, but this is a debug printing function so not sure if that's an issue, could think about a different container like `std::set` for `Visited` if so.

Reply:
I modified the code to avoid using MVisited.
Agree. I think there is no issue wasting time doing this vector research because this function should not be use in critical path but only for debugging as writing to a file is a time consuming operation.